python/src/opendp/context.py

```diff
@@ -90,6 +92,40 @@
         constructors[name[5:]] = constructor, is_partial
 
 
+def register(
+    constructor: Callable[..., Union[Transformation, Measurement]],
+    name: Optional[str] = None,
+):
+    """Register a constructor function to be used in the Context API.
+
+    If the constructor supports partial application (first two arguments are ``input_domain`` and ``input_metric``),
+    then the input domain and input metric are omitted when called via the Context API.
```

Comment:
Having these two parameters is the convention for `then_` functions, I think? Maybe allow either `make_` or `then_` prefix from the user, and our example should use `then_`?

If we get a function named `make_whatever`, but it has the right arguments, I don't think we need to make it a big error.

Reply:
The user will always have a `make_` function, but only sometimes (or rarely) have a then function, if they find the `to_then` function. I'm leaning towards keeping the registration simpler and having less flexibility. It would also introduce an ambiguity if `name` is set, unless I still detect based on the constructor name.